src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProductUpdateBar.xaml.cs

```diff
@@ -56,7 +60,11 @@ private void OnUpdateLinkClick(object sender, RoutedEventArgs e)
             UpdateStarting(this, EventArgs.Empty);
 
             // invoke with priority as Background so that our window is closed first before the Update method is called.
-            Dispatcher.BeginInvoke(new Action(_productUpdateService.Update), DispatcherPriority.Background);
+            NuGetUIThreadHelper.JoinableTaskFactory.WithPriority(VsTaskRunContext.UIThreadBackgroundPriority).Run(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
It seems `StartOnIdle` (together with possibly specifying your priority if the default isn't good enough) might be a more concise syntax since it can replace WithPriority, Run, and Switch.